nucypher/cli/characters/alice.py

```diff
@@ -11,171 +12,411 @@
 from nucypher.config.keyring import NucypherKeyring
 
 
-@click.command()
-@click.argument('action')
-@click.option('--dev', '-d', help="Enable development mode", is_flag=True)
-@click.option('--force', help="Don't ask for confirmation", is_flag=True)
-@click.option('--dry-run', '-x', help="Execute normally without actually starting the node", is_flag=True)
-@click.option('--teacher', 'teacher_uri', help="An Ursula URI to start learning from (seednode)", type=click.STRING)
-@click.option('--min-stake', help="The minimum stake the teacher must have to be a teacher", type=click.INT, default=0)
-@click.option('--discovery-port', help="The host port to run node discovery services on", type=NETWORK_PORT)
-@click.option('--controller-port', help="The host port to run Alice HTTP services on", type=NETWORK_PORT, default=AliceConfiguration.DEFAULT_CONTROLLER_PORT)
-@click.option('--federated-only', '-F', help="Connect only to federated nodes", is_flag=True)
-@click.option('--network', help="Network Domain Name", type=click.STRING)
+# Args (geth, provider_uri, federated_only, dev, pay_with, network, registry_filepath)
+def _admin_options(func):
+    @click.option('--geth', '-G', help="Run using the built-in geth node", is_flag=True)
+    @click.option('--provider', 'provider_uri', help="Blockchain provider's URI", type=click.STRING)
+    @click.option('--federated-only', '-F', help="Connect only to federated nodes", is_flag=True)
+    @click.option('--dev', '-d', help="Enable development mode", is_flag=True)
+    @click.option('--pay-with', help="Run with a specified account", type=EIP55_CHECKSUM_ADDRESS)
+    @click.option('--network', help="Network Domain Name", type=click.STRING)
+    @click.option('--registry-filepath', help="Custom contract registry filepath", type=EXISTING_READABLE_FILE)
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        return func(*args, **kwargs)
+    return wrapper
+
+
+# Args (geth, provider_uri, federated_only, dev, pay_with, network, registry_filepath, config_file, discovery_port,
+#       hw_wallet, teacher_uri, min_stake)
+def _api_options(func):
+    @_admin_options
+    @click.option('--config-file', help="Path to configuration file", type=EXISTING_READABLE_FILE)
```

Comment:
May be a good idea to make repeating options like this one a variable and then use it everywhere it's needed, i.e.

    config_file_option = click.option('--config-file', help="Path to configuration file", type=EXISTING_READABLE_FILE)

and

    @_admin_options
    @config_file_option
    ...etc

Reply:
I like this suggestion.

Reply:
👍  - Filed #1431 .